### ISSUE TITLE
Run `obfuscated django project` and got unknown opcode

### ISSUE THREAD

**Issue by weiztech** [dashingsoft/pyarmor#197]
I'm running my django project (successfully obfuscated) 

```INFO     1178 scripts have been obfuscated
INFO     Generating runtime files to dist/pytransform
INFO     Extract pytransform.key
INFO     Extract license.lic
INFO     Copying /env/local/lib/python2.7/site-packages/pyarmor/platforms/linux/x86_64/_pytransform.so
INFO     Generate runtime files OK
INFO     Obfuscate scripts with restrict mode on
INFO     Obfuscate 1178 scripts OK.
```
and got below error.

```
XXX lineno: 3, opcode: 122
Traceback (most recent call last):
  File "<manage.py>", line 4, in <module>
  File "<frozen manage>", line 3, in <module>
SystemError: unknown opcode
```

OS: Ubuntu - 16.04
Python 2.7.12
PyArmor Version 5.9.0(registered)

**Comment by weiztech** [comment]
FYI, removed `--advance 1`.
now i can run `manage.py`

**Comment by jondy** [comment]
Does it work to obfuscate a simple script `foo.py` with `--advanced 1`?
```
pyarmor obfuscate --advanced 1 --exact foo.py
python dist/foo.py
```

**Comment by weiztech** [comment]
> Does it work to obfuscate a simple script `foo.py` with `--advanced 1`?
> 
> ```
> pyarmor obfuscate --advanced 1 --exact foo.py
> python dist/foo.py
> ```

the script cannot be run  with `--advanced 1`. the error is the same like above.
without that params. it's working fine.

**Comment by jondy** [comment]
Run the obfuscated script with `-d`, what's the extra error messages?
```
python -d dist/foo.py
```

**Comment by weiztech** [comment]
> Run the obfuscated script with `-d`, what's the extra error messages?
> 
> ```
> python -d dist/foo.py
> ```

```
Initialize python patcher failed (1534)
Pytransform Error: The script is obfuscated in advanced mode but the runtime library doesn't support when running dist/test.py
```

**Comment by jondy** [comment]
Got it. This python dynamic library can't be patched by PyArmor.  I need check the python dynamic library to fix this issue, could you upload the python library used by python2.7 (maybe `libpython2.7.so`) in this machine?

**Comment by weiztech** [comment]
here's the file
[libpython2.7.so.zip](https://github.com/dashingsoft/pyarmor/files/4260549/libpython2.7.so.zip)

**Comment by jondy** [comment]
Could you also run this test script in target machine? 
``` python
from ctypes import cdll, byref, c_int, c_long, c_void_p, c_char_p, pythonapi, CFUNCTYPE


def test_memory_allocate():
    c = cdll.LoadLibrary(None)

    prototype = CFUNCTYPE(c_void_p, c_void_p, c_char_p)
    dlsym = prototype(('dlsym', c))

    prototype = CFUNCTYPE(c_void_p, c_long)
    malloc = prototype(('malloc', c))

    prototype = CFUNCTYPE(c_void_p, c_void_p, c_long, c_int, c_int, c_int, c_long)
    mmap = prototype(('mmap', c))

    prototype = CFUNCTYPE(c_int, c_void_p, c_long, c_long)
    posix_memalign = prototype(('posix_memalign', c))

    assert dlsym and malloc and mmap and posix_memalign

    n = 4096
    PROT_READ = 1
    PROT_WRITE = 2
    MAP_PRIVATE = 2
    MAP_ANONYMOUS = 0x20

    refaddr = dlsym(pythonapi._handle, "PyEval_EvalCode")
    print("PyEval_EvalCode at 0x%x" % refaddr)
    print("malloc return 0x%x" % malloc(n))
    print("mmap return 0x%x"
          % mmap(0, n, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0))
    print("mmap with refaddr return 0x%x"
          % mmap(refaddr, n, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0))

    addr = c_void_p(0)
    print("posix_memalign return %d" % posix_memalign(byref(addr), n, n))
    print("Allocate addr at 0x%x" % addr.value)


if __name__ == '__main__':
    test_memory_allocate()
```

**Comment by weiztech** [comment]
> ```python
> from ctypes import cdll, byref, c_int, c_long, c_void_p, c_char_p, pythonapi, CFUNCTYPE
> 
> 
> def test_memory_allocate():
>     c = cdll.LoadLibrary(None)
> 
>     prototype = CFUNCTYPE(c_void_p, c_void_p, c_char_p)
>     dlsym = prototype(('dlsym', c))
> 
>     prototype = CFUNCTYPE(c_void_p, c_long)
>     malloc = prototype(('malloc', c))
> 
>     prototype = CFUNCTYPE(c_void_p, c_void_p, c_long, c_int, c_int, c_int, c_long)
>     mmap = prototype(('mmap', c))
> 
>     prototype = CFUNCTYPE(c_int, c_void_p, c_long, c_long)
>     posix_memalign = prototype(('posix_memalign', c))
> 
>     assert dlsym and malloc and mmap and posix_memalign
> 
>     n = 4096
>     PROT_READ = 1
>     PROT_WRITE = 2
>     MAP_PRIVATE = 2
>     MAP_ANONYMOUS = 0x20
> 
>     refaddr = dlsym(pythonapi._handle, "PyEval_EvalCode")
>     print("PyEval_EvalCode at 0x%x" % refaddr)
>     print("malloc return 0x%x" % malloc(n))
>     print("mmap return 0x%x"
>           % mmap(0, n, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0))
>     print("mmap with refaddr return 0x%x"
>           % mmap(refaddr, n, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0))
> 
>     addr = c_void_p(0)
>     print("posix_memalign return %d" % posix_memalign(byref(addr), n, n))
>     print("Allocate addr at 0x%x" % addr.value)
> ```

PyEval_EvalCode at 0x4b9840
malloc return 0x2870570
mmap return 0x7f5e792cb000
mmap with refaddr return 0x7f5e792ca000
posix_memalign return 0
Allocate addr at 0x28a8000

**Comment by jondy** [comment]
I know the problem, it need to rebuild the dynamic library to fix this issue. I'll do that after I'm back to the office about 2 weeks later.

**Comment by jondy** [comment]
Fixed in v5.9.1
